### ISSUE TITLE
is_hooked inconsistency between codenode and project

### ISSUE THREAD

**Issue by Kyle-Kyle** [angr/angr#1406]
There are three(as far as I know) interfaces(CFGNode, CodeNode, angr.Project) where you can check whether an address is hooked.
And I found their results are different, sometimes confusing.
To reproduce:
~~~
import angr
proj = angr.Project('./CADET_00002')#angr-dev/binaries/tests/cgc
cfg = angr.analyses.CFGFast(normalize=True)
node = cfg.get_any_node(0xa000001)#should be the address of a syscall
print(node.is_syscall)# false # doesn't look right to me
print(node.is_simprocedure)# true # even more confusing
print(node.to_codenode().is_hook)# true # OK?
print(proj.is_hooked(node.addr))# false # ???
~~~

**Comment by rhelmot** [comment]
The only ones that look wrong to me are node.is_syscall and
codenode()/.is_hook. Syscalls are simprocedures that are not applied via
the normal hooking interface, but via the SimOS's syscall dispatching
mechanism.

**Comment by Kyle-Kyle** [comment]
I know, I'm just saying there is an inconsistency, not saying all of they are wrong

**Comment by rhelmot** [comment]
I know, I'm just drawing your attention to which ones you should fix.

**Comment by Kyle-Kyle** [comment]
Got it

**Comment by ltfish** [comment]
Seems like a bug to me. @Kyle-Kyle are you fixing it, or should I do it?

**Comment by Kyle-Kyle** [comment]
Resolved in [#1413](https://github.com/angr/angr/pull/1413)
